### ISSUE TITLE
Missing props.children in QueryRenderer render func

### ISSUE THREAD

**Issue by adeperio** [4Catalyzer/found-relay#124]
Hi all

I seem to be running into an issue when passing in a custom render method in one of my Route objects.

My route setup is shown below, and I am at location "/timers/detail"

![image](https://user-images.githubusercontent.com/1326430/36775994-61899aca-1cb8-11e8-9b48-6c443d3bdc3f.png)

The render method I pass into the route takes the same signature as the render method of the QueryRenderer. And my component mapped to that route looks like this:

![image](https://user-images.githubusercontent.com/1326430/36776679-c4169f92-1cba-11e8-965b-9e5b9f7d27af.png)

And I pass this into my routes like so:

![image](https://user-images.githubusercontent.com/1326430/36775897-f9f909ea-1cb7-11e8-9879-925132e67b1e.png)

Now a few things happen on the "/timers/detail" route:

1. In the Parent component ("timers"), when I try to access props.children (which I am expecting to be "detail")  in the render method of the parent, props.children is now null.
2. I've also tried props.route.children, which is not null, but the wrapping parent "timers" component still does not render
3. However, the child component "detail" actually renders, despite the props.children on the parent "timers" being null. 
4. Also just as strange, the parent, when in the location "/timers/detail" doesn't actually render at all.
5. However, when I navigate up a level to "timers", this component renders correctly again.
6. When using a regular React component render() method, instead of passing in a custom render method to the route, this all works correctly as expected. So the problem only occurs when passing in a custom render method to the route.

I'm not sure where I have went wrong with my setup here? Any clues as to how I can get the parent component to render correctly when in "timers/detail" 

Thanks all

**Comment by adeperio** [comment]
Hi all, I think I resolved the issue by doing a little refactor. Looks like the QueryRenderer render method doesn't like anything but a top level component passed into it. 

Seems like it won't render any of the JSX inside the root divs that you supply to it _within_ the render function itself. 

I got this to work by passing in a component directly as the root div, then passing the props data into that component. I think this is a similar scenario to error and loading states. 

In short, components that you specify in your Route that take a custom render definition, should be plain components just server to pass the QueryRenderer data back through to another component.

At least I think this is my understanding of how it is currently working. The following code ended up working for me in the end:

![image](https://user-images.githubusercontent.com/1326430/36779976-ce30eea0-1cc5-11e8-9359-7c8cdbd75dc4.png)

**Comment by taion** [comment]
The two-pass thing is unavoidable here; we need to call route `render` functions from parent to child in order to check for redirects or errors in the right order, but then we need to set up `children` separately from child to parent to get the values, so `render` should return an element that can get a `children` prop cloned in. (TBD maybe we want to support returning a function `(children) => element`).

Separately, your `AdminTimers` there isn't actually a React component. There's no need to specify `Component` if you're specifying a `render` method on the route unless you're using the component for code splitting.

**Comment by adeperio** [comment]
Thanks for the heads up @taion, that's actually good to know and get clarification on - I initially found that bit a little confusing with the render method. As someone who was looking at the code for the first time I assumed that the route's render method overrode the component's render method on the route. If that is meant to be synonymous with the function signature of QueryRenderer's render method might be more of a problem with Relay docs rather than found-relay, although would be handy if that was a little clearer.

And yep I actually ended up refactoring that render method out even further to a function in the same file rather than as statics on that component (Which wasn't really acting as a component as you pointed out).  Also removed the Component from the route as you suggested too.

So I ended up moving forward with something like this: 

![image](https://user-images.githubusercontent.com/1326430/36818184-123dc0b0-1d38-11e8-8735-66f2ec899c5e.png)
![image](https://user-images.githubusercontent.com/1326430/36817522-8c43bd4a-1d35-11e8-9a0b-66f0e2663af2.png)

**Comment by taion** [comment]
Looks good. If you do want to use code splitting at some point, you'd benefit from not defining the Route `render` function in the same bundle as the component itself, though.

**Comment by adeperio** [comment]
Ahh yep gotcha, I think code-splitting will start making its way in at some point so I'll try refactor that out too in preparation. 

Removing Component out of the Route kind of makes it less explicit the parent-child heirarchy between timers->detail components if the render function is in a seperate file, so I'll try get this over to the routes file instead. 

Might make for a meatier routes file, but at least the component used in the render method will be visible in there in the correct hierarchy position as it's intended children.

**Comment by anhang** [comment]
Hi @taion - I can no longer access `props.children` as in this example, in `0.4.0-alpha.6`. Has this feature been removed?

**Comment by taion** [comment]
It hasn't ever been possible to access `props.children` there. You either need to return a prop that will be cloned, or return a function from `render` per https://github.com/4Catalyzer/found#render
